snekbox/nsjail.py

```diff
@@ -19,18 +20,35 @@
 
 log = logging.getLogger(__name__)
 
+T = TypeVar("T")
+
 # [level][timestamp][PID]? function_signature:line_no? message
 LOG_PATTERN = re.compile(
     r"\[(?P<level>(I)|[DWEF])\]\[.+?\](?(2)|(?P<func>\[\d+\] .+?:\d+ )) ?(?P<msg>.+)"
 )
 
 
+def iter_lstrip(iterable: Iterable[T]) -> Generator[T, None, None]:
+    """Removes leading falsy objects from an iterable."""
+    it = iter(iterable)
+    for item in it:
+        if item:
+            yield item
+            break
+    yield from it
+
+
 def parse_files(
     fs: MemFS,
     files_limit: int,
     files_pattern: str,
 ) -> list[FileAttachment]:
-    """Parse files in a MemFS."""
+    """
+    Parse files in a MemFS.
+
+    Returns:
+        List of FileAttachments sorted lexically by path name.
+    """
```

Comment:
Okay this comment is a bit outdated now that I noticed you changed the docstring style. Still, I think this could just be changed to the one liner I suggested above.

Reply:
Refactored `parse_files` as an instance function `files_list` of `MemFS` in [8272f88](https://github.com/python-discord/snekbox/pull/159/commits/8272f88c3b50cc3912ae087b4f7eec506a5abaac). Made argument naming between `files` and `files_list` more coherent.